src/lib/core/overlay/overlay-directives.ts

```diff
@@ -150,20 +156,27 @@ export class ConnectedOverlayDirective implements OnDestroy {
       overlayConfig.backdropClass = this.backdropClass;
     }
 
-    overlayConfig.positionStrategy = this._getPosition();
+    overlayConfig.positionStrategy = this._setUpPosition();
 
     overlayConfig.direction = this.dir;
 
     return overlayConfig;
   }
 
+  /** Sets up the position strategy with its initial configuration. */
+  private _setUpPosition(): ConnectedPositionStrategy {
```

Comment:
It's a bit odd that there's `getPosition` and `setUpPosition`. Combine them into one method like `_createPositionStrategy`?

```ts
private _createPositionStrategy() {
  let originPoint = {originX: this.positions[0].overlayX, originY: this.positions[0].originY};
  let overlayPoint = {overlayX: this.positions[0].overlayX, overlayY: this.positions[0].overlayY};

  return this._overlay.position()
      .connectedTo(this.origin.elementRef, originPoint, overlayPoint)
      .withDirection(this.dir)
      .withOffsetX(this.offsetX)
      .withOffsetY(this.offsetY);
}
```